tests/test_submit.py

```diff
@@ -0,0 +1,408 @@
+    from src.submit import _create_submission_commit
+
+    repo, _branch = git_repo
+
+    # Create an untracked file
+    test_file = repo / "untracked.txt"
+    test_file.write_text("untracked content")
+
+    await _create_submission_commit(repo)
+
+    # Verify the submission commit was created
+    commit_message = git_command(repo, "log", "-1", "--pretty=%B", capture_output=True)
+    assert commit_message == (0, "SUBMISSION")
+
+    # Verify the untracked file is still present
+    status_code, status = git_command(repo, "status", "--porcelain", capture_output=True)
+    assert status_code == 0
+    assert status and "?? untracked.txt" in status
+
+
+@pytest.mark.asyncio
+async def test_git_push_successful(git_repo_with_remote: tuple[pathlib.Path, str]):
+    from src.submit import _git_push
+
+    repo, _branch = git_repo_with_remote
+
+    return_code, output = await _git_push(repo)
+    assert return_code == 0, output
+    assert output and "Everything up-to-date" in output
+
+
+@pytest.mark.asyncio
+async def test_git_push_fails_with_no_remote(git_repo: tuple[pathlib.Path, str]):
+    from src.submit import _git_push
+
+    repo, _branch = git_repo
+
+    return_code, output = await _git_push(repo)
+    assert return_code != 0, output
+    assert "No configured push destination" in output
+
+
+@pytest.mark.asyncio
+async def test_git_push_with_conflicts(
+    git_repo_with_remote: tuple[pathlib.Path, str], remote_repo: pathlib.Path
+):
+    """Test push failing when remote has diverged"""
+    from src.submit import _git_push
+
+    repo, _ = git_repo_with_remote
+
+    # Clone the remote to create divergent history
+    remote = remote_repo
+    other_clone = remote.parent / "other_clone"
+    git_command(remote.parent, "clone", str(remote), "other_clone")
+
+    # Create conflicting commit in other clone
+    conflict_file = other_clone / "conflict.txt"
+    conflict_file.write_text("conflict content")
+    git_command(other_clone, "config", "user.name", "Test User")
+    git_command(other_clone, "config", "user.email", "test@example.com")
+    git_command(other_clone, "add", "conflict.txt")
+    git_command(other_clone, "commit", "-m", "Conflicting commit")
+    git_command(other_clone, "push")
+
+    # Create local commit
+    local_file = repo / "local.txt"
+    local_file.write_text("local content")
+    git_command(repo, "add", "local.txt")
+    git_command(repo, "commit", "-m", "Local commit")
+
+    # Push should now fail
+    return_code, output = await _git_push(repo)
+    assert return_code != 0
+    assert "[rejected]" in output or "failed to push" in output
+
+
+@pytest.mark.asyncio
+async def test_check_git_repo_clean(
+    git_repo_with_remote: tuple[pathlib.Path, str], mocker
+):
+    from src.submit import _check_git_repo
+
+    repo, _ = git_repo_with_remote
+
+    echo_mock = mocker.patch("click.echo")
+    with mock.patch("click.confirm", return_value=True):
+        await _check_git_repo(repo)
```

Comment:
1. The mixture of `mocker` and `mock` here is strange, only need `mocker`
2. always use `autospec=True`